packages/winston-log-enricher/lib/createFormatter.js

```diff
@@ -13,86 +13,192 @@ module.exports = function createFormatter(newrelic, winston) {
-      // Due to Winston internals sometimes the error on the info object is a string or an
-      // empty object, and so the message property is all we have
-      const errorMessage = info.error.message || info.message || ''
-
-      info['error.message'] = truncate(errorMessage)
-      info['error.class'] =
-        info.error.name === 'Error' ? info.error.constructor.name : info.error.name
-      info['error.stack'] = truncate(info.error.stack)
-      info.message = truncate(info.message)
-
-      // Removes additional capture of stack to reduce overall payload/log-line size.
-      // The server has a maximum of ~4k characters per line allowed.
-      delete info.trace
-      delete info.stack
+    if (isMetricsEnabled(config)) {
+      incrementLoggingLinesMetrics(info, newrelic)
     }
 
-    if (info.timestamp) {
-      newrelic.shim.logger.traceOnce(
-        'Overwriting `timestamp` key; assigning original value to `original_timestamp`.'
-      )
-      info.original_timestamp = info.timestamp
-    }
-    info.timestamp = Date.now()
-
-    const metadata = newrelic.getLinkingMetadata(true)
-
-    // Add the metadata to the info object being logged
-    Object.keys(metadata).forEach((m) => {
-      info[m] = metadata[m]
-    })
-
-    const config = newrelic.shim.agent.config
-
-    // TODO: update the peerdep on the New Relic repo and thereby
-    // remove check for existence of application_logging config item
-    if (
-      config.application_logging &&
-      config.application_logging.enabled &&
-      config.application_logging.metrics.enabled
-    ) {
-      const levelLabel = info.level
-      newrelic.shim.agent.metrics.getOrCreateMetric('Logging/lines').incrementCallCount()
-      newrelic.shim.agent.metrics
-        .getOrCreateMetric(`Logging/lines/${levelLabel}`)
-        .incrementCallCount()
-    }
-
-    const forwardingEnabled =
-      config.application_logging &&
-      config.application_logging.enabled &&
-      config.application_logging.forwarding &&
-      config.application_logging.forwarding.enabled
-
-    if (newrelic.agent.logs && forwardingEnabled) {
-      const segment = newrelic.shim.getActiveSegment()
-
-      // Priority could be null even if we have a transaction, as the
-      // priority is usually set at the end of transactions, DT being
-      // a notable exception. Otherwise, matching log priority to the
-      // transaction priority can increase the odds of both being
-      // sampled.
-      const priority = segment && segment.transaction && segment.transaction.priority
-      newrelic.agent.logs.add(info, priority)
+    if (isLocalDecoratingEnabled(config)) {
+      const metadata = newrelic.getLinkingMetadata(true)
+      info.message += formatLinkingMetadata(metadata)
+    } else if (isLogFowardingEnabled(config, newrelic)) {
+      const metadata = newrelic.getLinkingMetadata(true)
+      reformatLogLine(info, metadata, newrelic)
+      newrelic.agent.logs.add(info)
```

Comment:
Doesn't this require a priority? Or did we determine it doesn't after all?

Reply:
`newrelic.agent.logs` is now a custom class that handles the priority on its own.

https://github.com/newrelic/node-newrelic/pull/1174/files#diff-44e3efd3723af2a04792734d1602a1816d5e2d03708721c76e586eb8182b25ccR16